File: qqbot-persistence/src/main/java/com/mieai/qqbot/persistence/outbox/OutboxPage.java
package com.mieai.qqbot.persistence.outbox;

import java.util.List;
import java.util.Objects;
import java.util.Optional;

/** One immutable page of Outbox jobs and an optional cursor for the next page. */
public record OutboxPage(List<OutboxJob> jobs, Optional<String> nextCursor) {
    public OutboxPage {
        jobs = List.copyOf(Objects.requireNonNull(jobs, "jobs must not be null"));
        nextCursor = Objects.requireNonNull(nextCursor, "nextCursor must not be null");
    }
}

File: qqbot-persistence/src/main/kotlin/com/mieai/qqbot/persistence/outbox/OutboxPage.kt
package com.mieai.qqbot.persistence.outbox

/** One immutable page of Outbox jobs and an optional cursor for the next page. */
data class OutboxPage private constructor(
    val jobs: List<OutboxJob>,
    val nextCursor: String?,
    private val normalized: Boolean,
) {
    constructor(jobs: List<OutboxJob>, nextCursor: String?) : this(
        jobs.toList(),
        nextCursor,
        true,
    )
}
